src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Store.cs

```diff
@@ -128,6 +128,9 @@ public void Add(X509Certificate2 certificate)
         {
             if (certificate == null)
                 throw new ArgumentNullException(nameof(certificate));
+            
+            if (certificate.Handle == IntPtr.Zero)
+                throw new ArgumentNullException(nameof(certificate));
```

Comment:
The exception order is a bit off, compared to .NET Framework.

https://referencesource.microsoft.com/#System/security/system/security/cryptography/x509/x509store.cs,198

Looks like  cert null check, store open check (the next line), and then check that the store is open again (redundant), and that the cert handle isn't null (which is what I said before was the state we were in, but I think I was wrong), and finally that the cert handle isn't invalid.

That said, I think that to match netfx we need to move this down one check and that it should be 
`throw new CryptographicException(SR.GetString(SR.Cryptography_InvalidHandle), "pCertContext");`

Assuming I figured out how to trigger a netfx run, we can find out :smile:.